src/main/java/world/bentobox/bentobox/managers/IslandsManager.java

```diff
@@ -266,7 +274,13 @@ public void deleteIsland(@NonNull Island island, boolean removeBlocks) {
         if (removeBlocks) {
             // Remove island from the cache
             islandCache.deleteIslandFromCache(island);
-            // Remove the island from the database
+            // Log the deletion (it shouldn't matter but may be useful)
+            island.log(new LogEntry.Builder("DELETED").build());
+            // Set the delete flag which will prevent it from being loaded even if database deletion fails
+            island.setDeleted(true);
+            // Save the island
+            handler.saveObject(island);
+            // Delete the island
```

Comment:
I know I save the file right before deleting it, but this is just "insurance" so that if for some reason the file doesn't delete, it won't be accidentally loaded and we can track why.